addons/l10n_ro_edi/models/account_edi_xml_ubl_ciusro.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, _
+
+
+SECTOR_RO_CODES = ('SECTOR1', 'SECTOR2', 'SECTOR3', 'SECTOR4', 'SECTOR5', 'SECTOR6')
+
+
+class AccountEdiXmlUBLRO(models.AbstractModel):
+    _inherit = "account.edi.xml.ubl_bis3"
+    _name = "account.edi.xml.ubl_ro"
+    _description = "CIUS RO"
+
+    def _export_invoice_filename(self, invoice):
+        return f"{invoice.name.replace('/', '_')}_cius_ro.xml"
+
+    def _get_partner_address_vals(self, partner):
+        # EXTENDS account.edi.xml.ubl_bis3
+        vals = super()._get_partner_address_vals(partner)
+
+        if partner.state_id:
+            vals["country_subentity"] = partner.country_code + '-' + partner.state_id.code
+
+        return vals
+
+    def _get_invoice_tax_totals_vals_list(self, invoice, taxes_vals):
+        # EXTENDS account.edi.xml.ubl_bis3
+        vals_list = super()._get_invoice_tax_totals_vals_list(invoice, taxes_vals)
+
+        if invoice.currency_id.name != 'RON':
+            # add new TaxTotal in RON
+            from_currency = invoice.currency_id
+            ron_currency = self.env.ref("base.RON")
+            from_amount = vals_list[0]['tax_amount']
+            to_amount = from_currency._convert(from_amount, ron_currency, invoice.company_id, invoice.date, False)
```

Comment:
Noooo! Everything you need is already in `taxes_vals` in both currencies. 
Take a look at `_get_invoice_tax_totals_vals_list` in 2.0 template. Just remove the trailing _currency and you should get the correct result

Reply:
I'm not sure about this: if you have an invoice which is not in RON currency (let's say it's in EUR), you will have every amounts in the xml in EUR except the TaxAmount in the TaxTotals ? Is it what is expected ?

Reply:
@JulienVR We will still have the normal TaxTotal in EUR but add another TaxTotal in RON, so in total there will be 2 TaxTotal. [Example here](https://github.com/odoo/odoo/pull/130617/files/26b1e453633c889c8312a62ddbe510de5a431ef1#diff-dcc450ce3734bd538eb8f1fdf70003f4dd27c2fde910b614d4f19dcc1673c4b1R107-R123)